actions/rules/action_state_to_metastate_test.go

```diff
@@ -0,0 +1,245 @@
+
+func TestSuiteActionStateToMetastate(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &ActionStateToMetastateSuite{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+type ActionStateToMetastateSuite struct {
+	gormtestsupport.DBTestSuite
+}
+
+func (s *ActionStateToMetastateSuite) TestActionExecution() {
+	// Note on the fixture: by default, the created board is attached to the type group
+	// with the same index. The columns on each board are as follows:
+	//   0 "New" "mNew"
+	//   1 "In Progress" "mInprogress"
+	//   2 "Resolved" "mResolved"
+	//   3 "Approved" "mResolved"
+	// All columns are set to the "BidirectionalStateToColumn" rule. The type has
+	// the following states/metastates:
+	//   "new"         "mNew"
+	//   "open"        "mOpen"
+	//   "in progress" "mInprogress"
+	//   "resolved"    "mResolved"
+	//   "closed"      "mClosed"
+	fxtFn := func(t *testing.T) *tf.TestFixture {
+		// this sets up a work item in state "new" with matching metastate "mNew" and the corresponding column 0 ("New")
+		// this represents a single work item in an ordered "new" state.
+		return tf.NewTestFixture(t, s.DB, tf.CreateWorkItemEnvironment(), tf.WorkItemBoards(1), tf.WorkItems(1, func(fxt *tf.TestFixture, idx int) error {
+			fxt.WorkItems[idx].Fields[workitem.SystemState] = "new"
+			fxt.WorkItems[idx].Fields[workitem.SystemMetaState] = "mNew"
+			fxt.WorkItems[idx].Fields[workitem.SystemBoardcolumns] = []string{
+				fxt.WorkItemBoards[idx].Columns[0].ID.String(),
+			}
+			return nil
+		}))
+	}
+	// test the fixture creation.
+	fxt := fxtFn(s.T())
+	require.NotNil(s.T(), fxt)
+	require.Len(s.T(), fxt.WorkItems, 1)
+	require.Equal(s.T(), "mNew", fxt.WorkItems[0].Fields[workitem.SystemMetaState])
+	require.Equal(s.T(), "new", fxt.WorkItems[0].Fields[workitem.SystemState])
+	require.Equal(s.T(), fxt.WorkItemBoards[0].Columns[0].ID.String(), fxt.WorkItems[0].Fields[workitem.SystemBoardcolumns].([]interface{})[0])
+
+	s.T().Run("updating the state for an existing work item", func(t *testing.T) {
+		fxt := fxtFn(t)
+		// set the state to "in progress" and create a changeset.
+		fxt.WorkItems[0].Fields[workitem.SystemState] = "in progress"
+		contextChanges := []convert.Change{
+			{
+				AttributeName: workitem.SystemState,
+				OldValue:      "new",
+				NewValue:      "in progress",
+			},
+		}
+		// run the test.
+		action := ActionStateToMetaState{
+			Db:     s.GormDB,
+			Ctx:    s.Ctx,
+			UserID: &fxt.Identities[0].ID,
+		}
+		var convertChanges []convert.Change
+		// note: changing the state does not require a configuration.
+		afterActionWI, convertChanges, err := action.OnChange(*fxt.WorkItems[0], contextChanges, "", &convertChanges)
+		require.Nil(t, err)
+		require.Len(t, convertChanges, 2)
+		// check metastate validity.
+		require.Equal(t, "system.metastate", convertChanges[0].AttributeName)
+		require.Equal(t, "mNew", convertChanges[0].OldValue)
+		require.Equal(t, "mInprogress", convertChanges[0].NewValue)
```

Comment:
This should be a constant as well and why is it `mInprogress` and not `mInProgress` or even better `in_progress`?

Reply:
Same as above, literals in tests show issues in constants.

Reply:
No it shouldn't.

Reply:
Yes it should. We're building this to fail, but anyway, outdated. Already fixed.